src/ui/component/dropDownMenu/dropDownMenu.tsx

```diff
@@ -0,0 +1,159 @@
+import { useAppStore } from '@/ui/store/appStore'
+import { Icon } from '@/ui/component/icon/icon'
+import classnames from 'classnames'
+import './dropDownMenu.scss'
+
+type OptionID = string
+
+export type Option = {
+  id: OptionID
+  title: string
+  value: OptionID[]
+}
+
+type DropDownMenuPage = {
+  title: string
+  menuOptions: Option[]
+}
+
+type DropDownMenuProps = {
+  options: Option[]
+  onSelect: (
+    selectedOptionValue: OptionID[],
+    setSelectedTitle: Dispatch<SetStateAction<string>>
+  ) => void
+  placeholder?: string
+}
+
+// eslint-disable-next-line max-lines-per-function
+export const DropDownMenu: FC<DropDownMenuProps> = ({ options, onSelect, placeholder = '' }) => {
+  const [menuPages, setMenuPages] = useState<DropDownMenuPage[]>([])
+  const [selectedTitle, setSelectedTitle] = useState<string>(placeholder)
+  const [focusedMenuPageIndex, setFocusedMenuPageIndex] = useState<number>(0)
+  const menuRef = useRef<HTMLDivElement | null>(null)
+  const menuPageRefs = useRef<HTMLDivElement[]>([])
+  const theme = useAppStore(state => state.theme)
+  const { t } = useTranslation('common')
+  const firstMenuPage = useMemo(() => options.filter(({ id, value }) => id === value[0]), [options])
+
+  const toggleMenu = useCallback(() => {
+    menuPages.length
+      ? setMenuPages([])
+      : setMenuPages([{ title: selectedTitle, menuOptions: firstMenuPage }])
+    setFocusedMenuPageIndex(0)
+  }, [menuPages.length, selectedTitle, firstMenuPage])
+
+  const clickOutsideHandler = useCallback(() => {
+    setMenuPages([])
+    setFocusedMenuPageIndex(0)
+  }, [])
+
+  useOnClickOutside(menuRef, clickOutsideHandler)
+
+  const selectOption = useCallback(
+    (selectedValue: OptionID[]) => {
+      onSelect(selectedValue, setSelectedTitle)
+      toggleMenu()
+    },
+    [onSelect, toggleMenu]
+  )
+
+  const slideMenu = useCallback(
+    (menuPage: number, menu?: DropDownMenuPage) => {
+      menuPageRefs.current[menuPage].scrollIntoView({
+        behavior: 'smooth',
+        block: 'nearest',
+        inline: 'center'
+      })
+      menu && setMenuPages([...menuPages.slice(0, menuPage), menu])
+      setFocusedMenuPageIndex(menuPage)
+    },
+    [menuPages]
+  )
+
+  const handleOptionClick = useCallback(
+    (options: Option[], index: number) => () => {
+      options.length === 1
+        ? selectOption(options[0].value)
+        : slideMenu(index + 1, { title: options[0].title, menuOptions: options.slice(1) })
+    },
+    [selectOption, slideMenu]
+  )
+
+  const handleMenuBack = useCallback(
+    (index: number) => () => {
+      slideMenu(index - 1)
+    },
+    [slideMenu]
+  )
+
+  const handleRef = useMemo(
+    () => (index: number) => (ref: HTMLDivElement) => (menuPageRefs.current[index] = ref),
+    []
+  )
+
+  const selectTitle = useMemo(
+    () => (menuPages.length ? menuPages[focusedMenuPageIndex].title : selectedTitle),
+    [focusedMenuPageIndex, menuPages, selectedTitle]
+  )
```

Comment:
Nice tweet and interesting point! 🙂 

Yes you're right that this value might be better not to memoize, but to be sure of this you'd really have to ask the question every time you make a function or variable how it will perform, accounting for parent and component re-renders. This can become quite taxing.

I've a tendance to agree with the[ second article](https://attardi.org/why-we-memo-all-the-things/) that memoizing functions and values is a sane default.

Maybe we could optimize the memoization for the entire app in a separate ticket in the future?